File: CSharp/TransmissionEvent.cs
using CanEasy;

namespace MyApplication
{
    public class Application : BaseApplication
    {
        protected override void OnStarting()
        {
            var msg = new CanEasy.DatabaseItem();
            msg.StringRef = "Msg:Message";
            msg.OnTransmission += (trs) =>
            {
                var sData = System.BitConverter.ToString((byte[])trs.Data.Array);
                CanEasy.MakeReport($"Message Id=0x{trs.MsgID.Plain.ToString("x2")}, Rx={trs.Received}, Data={sData}" , ReportType.ReportTypeInformation);
            };
            msg.TransmissionEvent.Active = true;
        }
    }
}

File: Python/TransmissionEvent.py
from caneasylib import *
import win32com.client as win32com

# Get CanEasyApplication instance
caneasy:ICanEasyApplication = getCanEasy()

class TrsEvent:
    def OnTransmission(self, _trs):
        trs = ITransmissionData(_trs)
        caneasy.MakeReport(f"Message Id={hex(trs.MsgID.Plain)}, Rx={trs.Received}, Data={bytes(trs.Data.Array).hex()}" , constants.ReportTypeInformation);

@entry
class Application(BaseApplication):
    def OnStarting(self):
        msg = caneasy.CreateDatabaseItem()
        msg.StringRef = "Msg:Message"
        win32com.WithEvents(msg, TrsEvent)
        msg.TransmissionEvent.Active = True
